src/python/zigopt/authentication/login.py

```diff
@@ -26,7 +26,7 @@ def right_code(user: User, code: str) -> bool:
     user.email_verification_timestamp
     and (now - user.email_verification_timestamp) < VERIFICATION_THRESHOLD
     and user.hashed_email_verification_code
-    and password_matches(code, user.hashed_email_verification_code or "")
+    and password_matches(code, user.hashed_email_verification_code)
```

Comment:
is the default for a "non"code `None` or `""`; if the former, will this still run correctly or spew an error?

Reply:
The error from coverity was pointing out that in the following statement:
```python
user.hashed_email_verification_code
    and password_matches(code, user.hashed_email_verification_code or "")
```
if the `password_matches` function is reached then `user.hashed_email_verification_code` is necessarily truthy, therefore the fallback `""` will never be used.